### DISCUSSION TITLE
I keep getting internal server error in Supabase when I click the google single sign on button

### DISCUSSION THREAD

**Discussion by pariszhuang123** [supabase/supabase#26820] [Questions] [answered]
Hi all, 

I have not being able to login via Google Single Sign on to Supabase. 

**Steps and logging in app**
This are the steps and logging I have done when I ran my flutter APK in android studio:
- FIrst they load the configuration for the dev environment. 
- Then the configuration was loaded successfully
- After that the app fetched the supabase URL
- Then the app fetched supabase Anon Key
- Supabase initi completed instance of "Supabase"
   - I/flutter ( 4438): ***** SupabaseDeepLinkingMixin startAuthObserver
I/flutter ( 4438): **** onAuthStateChange: AuthChangeEvent.initialSession
- Then I trigger the Google Sign in Process
    -> They fetched the web client ID 
     -> Fetched the iOS Client ID 
     -> Received the access token 
     -> Receive the ID Token 
     -> Failed to decode the access token (Failed assertion: line 10 pos 10: 'parts.length == 3': is not true.) [I think it is because of opaque token]
      -> Successful in decoding ID Token
    - Sign in with Supabase
       ->Error during Google Sign In: AuthException(message: Internal Server Error, statusCode: 400)

**What is shown in Supabase**
The error is: ID token does not have a at_hash claim, access_token parameter is unused

This results in the internal server error: {"component":"api","error":"500: Database error saving new user","level":"info","method":"POST","msg":"server_error: Internal Server Error","path":"/token","referer":"http://localhost:3000","remote_addr":"118.92.241.32","time":"2024-05-28T08:22:53Z","timestamp":"2024-05-28T08:22:53Z"}

**What did i validate **
I have validated that my webclientID, iOSClientID and AndroidClientID is accurate. Not too sure what are thte problems that I am facing. 

**Does anybody knows why is this happening**

**Comment by pariszhuang123** [comment]
While I am doing further investigation, they keep saying I am missing "at_hash" claims in my id_token. Is it a must to have "at_hash" claims for google sign in?

**Reply by javierlobra** [reply]
Hi @pariszhuang123,

I am getting the same message regarding the missing "at_hash" claims in my id_token. Have you found any solution? I think it might be related to the fact that I am intermittently getting logged out of the app.

**Reply by pariszhuang123** [reply]
@javierlobra Supabase does not need the "at_hash" claims. The issue was that I was trying to complicate the google sign in method. 

What I had previously done, was replicate the process from this youtube video and it works. 
https://www.youtube.com/watch?v=utMg6fVmX0U&t=482

**Comment by Mykyta-Chernenko** [comment]
I have exactly the same issue, around a week ago I started getting OnboardingAuthGoogleError for some of the users in my Flutter app
 
```
      final GoogleSignIn googleSignIn = GoogleSignIn(
        clientId: iosClientId,
        serverClientId: webClientId,
      );
      final GoogleSignInAccount? googleUser = await googleSignIn.signIn();
      final googleAuth = await googleUser?.authentication;
      final accessToken = googleAuth?.accessToken;
      final idToken = googleAuth?.idToken;
      final user = await supabase.auth.signInWithIdToken(
        provider: OAuthProvider.google,
        idToken: idToken ?? "",
        accessToken: accessToken,
      );
      ```
      
{"error":"server_error","error_description":"Internal Server Error"}
AuthException(message: AuthRetryableFetchError,

my supabase auth logs show

{"component":"api","level":"info","method":"POST","msg":"ID token does not have a at_hash claim, access_token parameter is unused.","path":"/token","referer":"http://localhost:3000","remote_addr":"109.43.114.255","request_id":"8938938285e7928d-FRA","time":"2024-06-14T07:23:39Z"}

**Reply by Mykyta-Chernenko** [reply]
It seems that when I provide accessToken at_hash must be present in idToken? Is it possible to ignore accessToken when at_hash is not present?

**Reply by Mykyta-Chernenko** [reply, marked as answer]
Sorry for the confusion, the issue was unrelated to at_hash, it's just a log statement.

What happened is that the user indeed couldn't get created because I had a trigger on it that didn't execute properly

@pariszhuang123 , check the logs of the postgres, you will probably see some execution error when the user is created
